### ISSUE TITLE
Lib assumes first response is for last request (race condition)

### ISSUE THREAD

**Issue by erikmack** [memcachier/memjs#9]
Hello, and thank you for memjs!

To reproduce:
1. Run memcached on remote host across WAN (latency may be required, not sure)
2. In the node shell, run

``` javascript
var memjs = require('memjs'),client = memjs.Client.create();

client.get('hello',function(err,val,extras) {console.log(err + ', ' + val)});client.set('foo','bar',function(err,success) {console.log(err + ', ' + success);});
```

Assuming key 'hello' does not exist already

Expect:
'Key not found',null  // from get request
null,true  // from set request

Actual:

``` javascript
> client.get('hello',function(err,val,extras) {console.log(err + ', ' + val)});client.set('foo','bar',function(err,success) {console.log(err + ', ' + success);});
undefined
> null, null
MemJS SET: Key not found
Error: MemJS SET: Key not found, null
```

The 'get' response (with opcode 0 and status 1 = key not found) is executing in the callback for the 'set' request.  

In other words, the last callback registered will be executed upon receipt of the first outstanding response.

It may be necessary to correlate requests with response callbacks in some kind of data structure:

``` javascript
callbacks[opcode][key] = onecallback;
```

Then when a response arrives, the correct callback can be invoked.

I encountered this at the office today, and have reproduced it in a second environment at home.

Let me know if you have any questions, and keep up the good work!
